# Estruturas de decisão

<span style="color: red;"><b>Todos os exercícios são feitos partindo-se do pressuposto de que todas as entradas são dadas de forma correta. Casos limite não mencionados no enunciado não são abordados porque não fazem parte do exercício.</b></span>

#### 1. Faça um Programa que peça dois números e imprima o maior deles.

In [7]:
num1 = int(input('Digite o 1º número: '))
num2 = int(input('Digite o 2º número: '))

if num1 > num2:
    resultado = num1
    print (f'O 1º número é maior que 1º. Número digitado:{resultado}')
elif num1 == num2:
    print(f'Os dois números digitados são iguais')
    
else:
    resultado = num2
    print(f'O 2º número é maior que 1º. Número digitado: {resultado}')

Digite o 1º número: 1
Digite o 2º número: 1
Os dois números digitados são iguais


#### 2. Faça um Programa que peça um valor e mostre na tela se o valor é positivo ou negativo.

In [18]:
valor = int(input('Digite o valor: '))

if valor > 0:
    resultado = 'positivo'
elif valor == 0:
    resultado = 'neutro'
else:
    resultado = 'negativo'  
    
print (f'O número {valor} é {resultado}.')
print('FIM!')


Digite o valor: -3
O número -3 é negativo.
FIM!


#### 3. Faça um Programa que verifique o estado civil de uma pessoa. Se a letra digitada é "C" (Casado), "S" (Solteiro), "D" (Divorciado), "V" (Viúvo) ou "O" (outros). Conforme a letra escrita pelo usuário seu programa deve escrever o estado civil, exemplo:

Usuário digita: C

Seu programa deve responder:
C - Casado

In [32]:
estado_civil = str(input('Digite a letra referente a seu estado civil: '))

if estado_civil == 'C':
    resultado = "Casado"
elif estado_civil == 'S':
    resultado = "Solteiro"
elif estado_civil == 'D':
    resultado = "Divorciado"
elif estado_civil == 'V':
    resultado = "Viúvo"
else:
    resultado = "Outros"
print(f'Resultado {resultado}')

Digite a letra referente a seu estado civil: SDAFA
Resultado Outros


#### 4. Faça um Programa que verifique se o e-mail digitado faz parte dos e-mails de spam.

In [35]:
email = str(input('Digite seu e-mail:' ))

if email in "fulano@gmail.com,beltrano@gmail.com,ciclano@gmail.com":
    print (f'E-mail descartado, o {email} é um SPAM.')
else:
    print(f'E-mail: {email} foi cadastrado com sucesso.')
#emails_spam = "fulano@gmail.com,beltrano@gmail.com,ciclano@gmail.com"


Digite seu e-mail:thiarly.cavalcante@live.com
E-mail: thiarly.cavalcante@live.com foi cadastrado com sucesso.


#### 5. Faça um programa para a leitura de duas notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e apresentar:

##### A mensagem "Aprovado", se a média alcançada for maior ou igual a sete;
##### A mensagem "Reprovado", se a média for menor do que sete;
##### A mensagem "Aprovado com Distinção", se a média for igual a dez.

In [3]:
nota1 = float(input('Digita a 1º nota: '))
nota2 = float(input('Digite a 2º nota: '))

media = (nota1 + nota2) / 2

if media >= 7:
    if media < 10:
        print('Aprovado!')
    if media == 10:
        print('Aprovado com Distinção!')
else:
    print ('Reprovado!')
    


Digita a 1º nota: 10
Digite a 2º nota: 10
Aprovado com Distinção!


#### 6. Faça um Programa que leia o orçamento de 3 empresas e mostre o maior deles.

In [47]:
orcamento1 = (input('Digite o 1º orçamento: '))
orcamento2 = (input('Digite o 2º orçamento: '))
orcamento3 = (input('Digite o 3º orçamento: '))

if orcamento1 and orcamento2 and orcamento3:
    orcamento1 = int (orcamento1)
    orcamento2 = int (orcamento2)
    orcamento3 = int (orcamento3)
    
    if orcamento1 > orcamento3 and orcamento1 > orcamento2:
        print(f'O 1º orçamento é maior com valor R$: {orcamento1}.')
    if orcamento2 > orcamento3 and orcamento2 > orcamento1:
        print(f'O 2º orçamento é maior com valor R$:{orcamento2}.')
    if orcamento3 > orcamento2 and orcamento3 > orcamento1:
         print(f'O 3º orçamento é maior com valor R$:{orcamento3}.')
        
else:
    print (f'Preencha todos os campos.')
    
    
    
    #print(f'Os orçamentos 1, 2 e 3 são todos iguais com esse valor R$:{orcamento1} são iguais.')

Digite o 1º orçamento: 7
Digite o 2º orçamento: 7
Digite o 3º orçamento: 6


#### 7. Faça um Programa que leia três orçamentos e mostre o maior e o menor deles.

In [51]:
orcamento1 = (input('Digite o 1º orçamento: '))
orcamento2 = (input('Digite o 2º orçamento: '))
orcamento3 = (input('Digite o 3º orçamento: '))

if orcamento1 and orcamento2 and orcamento3:
    orcamento1 = float (orcamento1)
    orcamento2 = float (orcamento2)
    orcamento3 = float (orcamento3)
    
    #orçamento maior
    if orcamento1 < orcamento3 and orcamento1 < orcamento2:
        print(f'O 1º orçamento é maior com valor R$: {orcamento1}.')
    if orcamento2 < orcamento3 and orcamento2 < orcamento1:
        print(f'O 2º orçamento é maior com valor R$:{orcamento2}.')
    if orcamento3 < orcamento2 and orcamento3 < orcamento1:
         print(f'O 3º orçamento é maior com valor R$:{orcamento3}.')
            
            
    #orçamento menor
    if orcamento1 < orcamento3 and orcamento1 < orcamento2:
        print(f'O 1º orçamento é o menor valor R$: {orcamento1}.')
    if orcamento2 < orcamento3 and orcamento2 < orcamento1:
        print(f'O 2º orçamento é o menor valor R$:{orcamento2}.')
    if orcamento3 < orcamento2 and orcamento3 < orcamento1:
         print(f'O 3º orçamento é o menor valor R$:{orcamento3}.')
    
  
    
            
            
        
else:
    print (f'Preencha todos os campos.')

Digite o 1º orçamento: 7
Digite o 2º orçamento: 8
Digite o 3º orçamento: 5
O 3º orçamento é maior com valor R$:5.0.
O 3º orçamento é o menor valor R$:5.0.


#### 8. Faça um programa que pergunte o preço de três produtos e informe qual produto você deve comprar, sabendo que a decisão é sempre pelo mais barato.

In [64]:
produto1 = float(input('Informe o preço do 1º produto: '))
produto2 = float(input('Informe o preço do 2º produto: '))

if produto1 < produto2:
    #produto1 = float(produto1)
    #produto2 = float(produto2)
    print (f'Compre o produto 1º produto, está mais barato.')
else:
     print (f'Compre o produto 2º produto, está mais barato.')

Informe o preço do 1º produto: 45
Informe o preço do 2º produto: 688
Compre o produto 1º produto, está mais barato.


#### 9. Faça um Programa que leia três números e mostre-os em ordem decrescente.

In [80]:
num1 = int(input('Digite o primeiro número:'))
num2 = int(input('Digite o segundo número:'))
num3 = int(input('Digite o terceiro número:'))

if num1 < num2 and num1 < num3 and num2 < num3:
    print(f'Sequencia descrescente {num1}, {num2}, {num3}')
elif num1 < num2 and num1 < num3 and num3 < num2:
    print(f'Sequencia descrescente {num1}, {num3}, {num2}')

elif num2 < num1 and num2 < num3 and num1 < num3:
    print(f'Sequencia descrescente {num2}, {num1}, {num3}')
    
elif num2 < num1 and num2 < num3 and num3 < num1:
    print(f'Sequencia descrescente {num2}, {num3}, {num1}')

elif num3 < num1 and num3 < num2 and num1 < num2:
    print(f'Sequencia descrescente {num3}, {num1}, {num2}')
    
elif num3 < num1 and num3 < num2 and num2 < num1:
    print(f'Sequencia descrescente {num3}, {num2}, {num1}')



    
    
    
    
    
    
    
    
    
    
    
    

Digite o primeiro número:9
Digite o segundo número:6
Digite o terceiro número:3
Sequencia descrescente 3, 6, 9


#### 10. Faça um Programa que pergunte em que turno você estuda. Peça para digitar M-matutino ou V-Vespertino ou N- Noturno. Imprima a mensagem "Bom Dia!", "Boa Tarde!" ou "Boa Noite!" ou "Valor Inválido!", conforme o caso.

In [9]:
turno = input('Qual a letra do seu turno de Estudo M, V, N:')

if turno in "M, V, N":
    
    if turno == "M":
        print('BOM DIA!')
    if turno == "V":
        print('BOA TARDE!')
    if turno == "N":
        print('BOA NOITE!')
else: 
    print('Valor inválido')


Qual a letra do seu turno de Estudo M, V, N:N
BOA NOITE!


#### 11. As Organizações Tabajara resolveram dar um aumento de salário aos seus colaboradores e lhe contraram para desenvolver o programa que calculará os reajustes. Faça um programa que recebe o salário de um colaborador e o reajuste segundo o seguinte critério, baseado no salário atual:

##### salários até R\\$ 280,00 (incluindo) : aumento de 20% 

##### salários entre R\\$ 280,00 e R\\$ 700,00 : aumento de 15% 

##### salários entre R\\$ 700,00 e R\\$ 1500,00 : aumento de 10% 

##### salários de R\\$ 1500,00 em diante : aumento de 5% Após o aumento ser realizado, informe na tela: 

###### o salário antes do reajuste;

###### o percentual de aumento aplicado;

###### o valor do aumento;

###### o novo salário, após o aumento.

###### Obs: Não vamos nos preocupar tanto com a formatação dos números (nº de casas decimais, por exemplo, veremos isso no próximo módulo)

In [5]:
salario = float(input('Digite o salário: '))

if salario > 0:
    if salario <= 280:
        reajuste = 0.20
        aumento = 0.20 * salario
        novo_salario = aumento + salario
   
    elif salario <= 700:
        reajuste = 0.15
        aumento = 0.15 * salario
        novo_salario = aumento + salario
    
        
    elif salario <= 1500:
        reajuste = 0.10
        aumento = 0.10 * salario
        novo_salario = aumento + salario

        
    elif salario > 1500:
        reajuste = 0.05
        aumento = 0.05 * salario
        novo_salario = aumento + salario
   

    print ((f'Salário antes do reajuste: ${salario}, reajuste foi {reajuste}% o aumento foi {aumento}'))
    print (f'O novo salário: R$ {novo_salario}')
    

Digite o salário: 14700
Salário antes do reajuste: $14700.0, reajuste foi 0.05% o aumento foi 735.0
O novo salário: R$ 15435.0


#### 12 . Faça um programa para o cálculo de uma folha de pagamento, sabendo que os descontos são do Imposto de Renda, que depende do salário bruto (conforme tabela abaixo) e que o FGTS corresponde a 11% do Salário Bruto, mas não é descontado (é a empresa que deposita). O Salário Líquido corresponde ao Salário Bruto menos os descontos. O programa deverá pedir ao usuário o valor da sua hora e a quantidade de horas trabalhadas no mês.

Desconto do IR:<br>
Salário Bruto até 900 (inclusive) - isento<br>
Salário Bruto até 1500 (inclusive) - desconto de 5%<br>
Salário Bruto até 2500 (inclusive) - desconto de 10%<br>
Salário Bruto acima de 2500 - desconto de 20%<br>
Imprima na tela as informações, dispostas conforme o exemplo abaixo. No exemplo o valor da hora é 5 e a quantidade de hora é 220.

Salário Bruto: (5 * 220)        : R\\$ 1100,00<br>
(-) IR (5%)                     : R\\$   55,00<br>
(-) INSS ( 10%)                 : R\\$  110,00<br>
FGTS (11%)                      : R\\$  121,00<br>
Total de descontos              : R\\$  165,00<br>
Salário Liquido                 : R\\$  935,00<br>

###### Obs: Não vamos nos preocupar tanto com a formatação dos números (nº de casas decimais, por exemplo, veremos isso no próximo módulo)

In [15]:
valor_hora = float(input('Digite o valor da hora de trabalho R$: '))
trabalho_hora = float(input('Quantas horas trabalhou esse mês?:'))

salario = valor_hora * trabalho_hora
descontos = ir + inss
salario_liquido = salario - descontos

if salario <= 900:
    ir == 0
    inss = salario * 0.10
    fgts = salario * 0.11
    
elif salario <=1500:
    ir = salario * 0.05
    inss = salario * 0.10
    fgts = salario * 0.11
    
elif salario <=2500:
    ir = salario * 0.10
    inss = salario * 0.10
    fgts = salario * 0.11
    
else:
    ir = salario * 0.20
    inss = salario * 0.10
    fgts = salario * 0.11
    

print(f'Salário bruto R$ {salario}')
print(f'Desconto do IR R$: {ir}')
print(f'Desconto do INSS R$:{inss}')
print(f'Desconto do FGTS R$: {fgts}')
print (f'Total de descontos: R$ {descontos}, a empresa faz o desposito do FGTS.')
print(f'Salário Liquido R$: {salario_liquido}')



Digite o valor da hora de trabalho R$: 5
Quantas horas trabalhou esse mês?:220
Salário bruto R$ 1100.0
Desconto do IR R$: 55.0
Desconto do INSS R$:110.0
Desconto do FGTS R$: 121.0
Total de descontos: R$ 165.0, a empresa faz o desposito do FGTS.
Salário Liquido R$: 935.0


#### 13. Faça um Programa que leia um número e exiba o dia correspondente da semana. (1-Domingo, 2- Segunda, etc.), se digitar outro valor deve aparecer valor inválido.

In [48]:
valor = float(input('Digite o valor de 1 a 7: '))

if valor > 7:
    print('Valor inválido!')

elif valor == 1:
    print (f'1 = Domingo')
elif valor == 2:
    print (f'2 = Segunda')
elif valor == 3:
    print (f'3 = Terça-Feira')
elif valor == 4:
    print (f'4 = Quarta-Feira')
elif valor == 5:
    print (f'5 = Quinta-Feira')
elif valor == 6:
    print (f'6 = Sexta-Feira')
elif valor == 7:
    print (f'7 = Sábado')
    
else:
    print('Valor inválido!')


        

    



Digite o valor de 1 a 7: 0
Valor inválido!


#### 14. Faça um programa que lê as duas notas parciais obtidas por um aluno numa disciplina ao longo de um semestre, e calcule a sua média. Em seguida, mostre qual conceito o aluno teve. A atribuição de conceitos obedece à tabela abaixo:
<pre>
Média de Aproveitamento  Conceito
Entre 9.0 e 10.0        A
Entre 7.5 e 9.0         B
Entre 6.0 e 7.5         C
Entre 4.0 e 6.0         D
Entre 4.0 e zero        E
</pre>

In [5]:
nota1 = float(input('Digite a 1º nota:'))
nota2 = float(input('Digite a 2º nota:'))

media = (nota1 + nota2) / 2


if media <= 10:
    
    if media <= 4:
        conceito = "E"
    elif media <= 6:
        conceito = "D"
    elif media <= 7.5:
        conceito = "C"
    elif media <= 9:
        conceito = "B"
    elif media <= 10:
        conceito = "A"
   
    print (f'Conceito = {conceito}, valor da média {media}')

else:
    print('Redigite sua nota, valor inválido.')

    



Digite a 1º nota:9
Digite a 2º nota:235
Redigite sua nota, valor inválido.


#### 15. Você está construindo um calendário para controlar dias de trabalho a pedido do RH. Nessa construção, você vai precisar definir quais anos são bissextos e quais não são, para montar o calendário de forma correta. Faça um Programa que peça um número correspondente a um determinado ano e em seguida informe se este ano é ou não bissexto.
<pre>
Dica para determinar se um ano é bissexto: 
- São bissextos todos os anos múltiplos de 400, p.ex.: 1600, 2000, 2400, 2800...
- São bissextos todos os múltiplos de 4, exceto se for múltiplo de 100 mas não de 400, 
p.ex.: 1996, 2000, 2004, 2008, 2012, 2016, 2020, 2024, 2028...
- Não são bissextos todos os demais anos.<br>
ex1: 2004 é múltiplo de 4, mas não é múltiplo de 100, então é bissexto.
ex2: 2000 é múltiplo de 4, mas é múltiplo de 100, só que também é multiplo de 400, então é bissexto (porque todo ano múltiplo de 400 é bissexto, independente do resto).
ex3: 1900 é múltiplo de 4, é múltiplo de 100, mas não é múltiplo de 400, então não é bissexto

</pre>

Dica: lembre que: numero % 4 é o resto da divisão do número por 4, ex: 10 % 3 = 1 (já que 10/3 = 3 e resta 1)

In [10]:
ano = int(input('Ano: '))

if (ano%4==0 and ano%100!=0) or (ano%400==0):
    print('Bissexto')
else:
    print('Não é bissexto')

Ano: 2000
Bissexto


#### 16. Faça um Programa para leitura de três notas parciais de um aluno. O programa deve calcular a média alcançada por aluno e presentar:
<pre>
A mensagem "Aprovado", se a média for maior ou igual a 7, com a respectiva média alcançada;
A mensagem "Reprovado", se a média for menor do que 7, com a respectiva média alcançada;
A mensagem "Aprovado com Distinção", se a média for igual a 10.
</pre>

In [25]:
nota1 = float(input('Nota 1:'))
nota2 = float(input('Nota 2:'))
nota3 = float(input('Nota 3:'))

soma = (nota1 + nota2 + nota3) / 3

print (soma)

if soma == 10:
    print ('Aprovado com Distinção')
elif soma >= 7:
    print ('Aprovado')
else:
    print('Reprovado')




Nota 1:4
Nota 2:2
Nota 3:9
5.0
Reprovado


#### 17. João Papo-de-Pescador, homem de bem, comprou um microcomputador para controlar o rendimento diário de seu trabalho. Toda vez que ele traz um peso de peixes maior que o estabelecido pelo regulamento de pesca do estado de São Paulo (50 quilos) deve pagar uma multa de R$ 4,00 por quilo excedente. João precisa que você faça um programa que leia a variável peso (peso de peixes) e calcule o excesso. Gravar na variável excesso a quantidade de quilos além do limite e na variável multa o valor da multa que João deverá pagar. Imprima os dados do programa com as mensagens adequadas.

In [29]:
peso = float(input('Quanto kilos de peixe kg: '))

if peso > 50:
    excesso = peso - 50
    multa = excesso * 4
    print(f'Multa R$:{multa}')
else:
    print ('Peso dentro da norma.')

Quanto kilos de peixe kg: 51
Multa R$:4.0


#### 18. Faça um Programa para um caixa eletrônico. O programa deverá perguntar ao usuário a valor do saque e depois informar quantas notas de cada valor serão fornecidas. As notas disponíveis serão as de 1, 5, 10, 50 e 100 reais. O valor mínimo é de 10 reais e o máximo de 600 reais. O programa não deve se preocupar com a quantidade de notas existentes na máquina.
<pre>
Exemplo 1: Para sacar a quantia de 256 reais, o programa fornece duas notas de 100, uma nota de 50, uma nota de 5 e uma nota de 1;
Exemplo 2: Para sacar a quantia de 399 reais, o programa fornece três notas de 100, uma nota de 50, quatro notas de 10, uma nota de 5 e quatro notas de 1.
</pre>

Dica1: numero // 10 vai te dar como resposta a parte inteira da divisão do número por 10.<br>
Dica2: numero % 10 vai te dar o resto da divisão do número por 10.

In [1]:
valor = int(input('Informe o valor a ser sacado: '))

n100 = 0
n50 = 0
n10 = 0
n5 = 0
n1 = 0

if 10 <= valor <= 600:
    print('São necessárias:')
    if valor > 100:
        n100 = valor // 100
        print(n100, 'notas de 100')
    if valor % 100 >= 50:
        n50 = (valor - 100*n100) // 50
        print(n50, 'notas de 50')
    if valor % 50 >= 10:
        n10 = (valor - 100*n100 - 50*n50) // 10
        print(n10, 'notas de 10')
    if valor % 10 >= 5:
        n5 = (valor - 100*n100 - 50*n50 - 10*n10) // 5
        print(n5, 'notas de 5')
    if valor % 5 >= 1:
        n1 = valor - 100*n100 - 50*n50 - 10*n10 - 5*n5
        print(n1, 'notas de 1')
else:
    print('Informe um valor entre 10 e 600 reais')

Informe o valor a ser sacado: 125
São necessárias:
1 notas de 100
2 notas de 10
1 notas de 5


#### 19. Faça um programa que faça 5 perguntas para uma pessoa sobre um crime. As perguntas são:
<pre>
"Telefonou para a vítima?"
"Esteve no local do crime?"
"Mora perto da vítima?"
"Devia para a vítima?"
"Já trabalhou com a vítima?" 
O programa deve no final emitir uma classificação sobre a participação da pessoa no crime. Se a pessoa responder positivamente a 2 questões ela deve ser classificada como "Suspeita", entre 3 e 4 como "Cúmplice" e 5 como "Assassino". Caso contrário, ele será classificado como "Inocente".
</pre>

In [10]:
pergunta1 = str(input("Telefonou para a vítima?"))
pergunta2 = str(input("Esteve no local do crime?"))
pergunta3 = str(input("Mora perto da vítima?"))
pergunta4 = str(input("Devia para a vítima?"))
pergunta5 = str(input("Já trabalhou com a vítima?"))

if pergunta2 == "SIM":
    print('Suspeita')
elif pergunta3 and pergunta4 == "SIM":
    print('Cúmplice')
elif pergunta5 == "SIM":
    print('Assassino')
else:
    print ('Inocente')




Telefonou para a vítima?SIM
Esteve no local do crime?N
Mora perto da vítima?N
Devia para a vítima?N
Já trabalhou com a vítima?SIM
Assassino


#### 20. Um posto está vendendo combustíveis com a seguinte tabela de descontos:
<pre>
Álcool:
até 20 litros, desconto de 3% por litro
acima de 20 litros, desconto de 5% por litro
Gasolina:
até 20 litros, desconto de 4% por litro
acima de 20 litros, desconto de 6% por litro

Escreva um algoritmo que leia o número de litros vendidos, o tipo de combustível (codificado da seguinte forma: A-álcool, G-gasolina), calcule e imprima o valor a ser pago pelo cliente sabendo-se que o preço do litro da gasolina é R$ 2,50 o preço do litro do álcool é R$ 1,90.
</pre>

In [63]:
litros = float(input('Quantos litros foram vendidos:'))
tipo = str(input('Tipo de combustível, A-álcool, G-gasolina: '))
soma_a = (litros * 1.90)
soma_g = (litros * 2.50)
    
if tipo in "A, G":
    
    #CALCULO ALCOOL
    if tipo == "A" and litros <= 20:
        desconto = soma_a * 0.03
        valor = soma_a - desconto
        
    if tipo == "A" and litros > 20:
        desconto = soma_a * 0.05
        valor = soma_a - desconto
    
    
    
    #CALCULO GASOLINA
    if tipo == "G" and litros <= 20:
        desconto = soma_g * 0.04
        valor = soma_g - desconto
        
    if tipo == "G" and litros > 20:
        desconto = soma_g * 0.06
        valor = soma_g - desconto
        
       
    print(f'Valor a ser pago: R$ {valor}')
        
    
else:
    print('Gategoria de produto inválida!')




Quantos litros foram vendidos:50
Tipo de combustível, A-álcool, G-gasolina: G
Valor a ser pago: R$ 117.5


#### 21. Uma fruteira está vendendo frutas com a seguinte tabela de preços:
<pre>
                      Até 5 Kg           Acima de 5 Kg
Morango         R$ 2,50 por Kg          R$ 2,20 por Kg
Maçã            R$ 1,80 por Kg          R$ 1,50 por Kg
Se o cliente comprar mais de 8 Kg em frutas ou o valor total da compra ultrapassar R$ 25,00, receberá ainda um desconto de 10% sobre este total. Escreva um algoritmo para ler a quantidade (em Kg) de morangos e a quantidade (em Kg) de maças adquiridas e escreva o valor a ser pago pelo cliente.
</pre>

In [36]:
qtd_morango = float(input('Informe a quantidade de morango: '))
qtd_maca = float(input('Informe a quantidade de maçã: '))

if qtd_morango > 5:
    total_morango = 2.2 * qtd_morango
else:
    total_morango = 2.5 * qtd_morango
    
if qtd_maca > 5:
    total_maca = 1.5 * qtd_maca
else:
    total_maca = 1.8 * qtd_maca
    
if qtd_morango + qtd_maca > 8 or total_morango + total_maca > 25:
    total = (total_morango + total_maca) * 0.9
else:
    total = total_morango + total_maca
    
print('Total a pagar: R$', total)

0.57


#### 22. O Hipermercado Tabajara está com uma promoção de carnes que é imperdível. Confira:
<pre>
                      Até 5 Kg           Acima de 5 Kg
File Duplo      R$ 4,90 por Kg          R$ 5,80 por Kg
Alcatra         R$ 5,90 por Kg          R$ 6,80 por Kg
Picanha         R$ 6,90 por Kg          R$ 7,80 por Kg
Para atender a todos os clientes, cada cliente poderá levar apenas um dos tipos de carne da promoção, porém não há limites para a quantidade de carne por cliente. Se compra for feita no cartão Tabajara o cliente receberá ainda um desconto de 5% sobre o total da compra. Escreva um programa que peça o tipo e a quantidade de carne comprada pelo usuário e gere um cupom fiscal, contendo as informações da compra: tipo e quantidade de carne, preço total, tipo de pagamento, valor do desconto e valor a pagar.
</pre>

In [23]:
tipo_carne = int(input('1- File Duplo, 2- Alcatra, 3- Picanha:'))
quantidade = float(input('Quantidade Kg: '))
forma_pg = int(input('1- Dinheiro, 2- Debito, 3- Crédito, 4- Pix, 5- C. Tabajara:'))
                     
if tipo_carne == "File Duplo" and quantidade <= 5:
    total = quantidade * 4.90
elif tipo_carne == "File Duplo" and quantidade > 5:
    total = quantidade * 5.80
    
elif tipo_carne == "Alcatra" and quantidade <= 5:
    total = quantidade * 5.90
elif tipo_carne == "Alcatra" and quantidade > 5:
    total = quantidade * 6.80

elif tipo_carne == "Picanha" and quantidade <= 5:
    total = quantidade * 6.90
elif tipo_carne == "Picanha" and quantidade > 5:
    total = quantidade * 7.80

print('--- CUPOM FISCAL ---')
print(f'Tipo da carne: {tipo_carne}')
print(f'Quantidade:{quantidade}kg')
print(f'Preço total: R$ {total}')
print(f'Forma de pagamento: {forma_pg}')

desconto = 0.05 * total if pagamento == '5' else 0

print(f'Valor do desconto: R$ {desconto}')
print(f'Valor a pagar: R$ {total - desconto}')
    




1- File Duplo, 2- Alcatra, 3- Picanha:1
Quantidade Kg: 5
1- Dinheiro, 2- Debito, 3- Crédito, 4- Pix, 5- C. Tabajara:1
--- CUPOM FISCAL ---
Tipo da carne: 1
Quantidade:5.0kg


NameError: name 'total' is not defined

#### 23. Faça um Programa para uma loja de tintas. O programa deverá pedir o tamanho em metros quadrados da área a ser pintada. Considere que a cobertura da tinta é de 1 litro para cada 6 metros quadrados e que a tinta é vendida em latas de 18 litros, que custam R\\$ 80,00 ou em galões de 3,6 litros, que custam R\\$ 25,00.

In [9]:
area = float(input('Informe o tamanho da área a ser pintada em m²: '))
litros_tinta = area / 6



print (f'Para a essa área {area}m² é necessário {litros_tinta}L de tinta.')




Informe o tamanho da área a ser pintada em m²: 50
Para a essa área 50.0m² é necessário 8.333333333333334L de tinta.


##### Informe ao usuário as quantidades de tinta a serem compradas e os respectivos preços em 3 situações.

Dica: lembre dos operadores // e % mostrados em exercícios anteriores<br>
Dica1: numero // 10 vai te dar como resposta a parte inteira da divisão do número por 10.<br>
Dica2: numero % 10 vai te dar o resto da divisão do número por 10.

##### 1. Comprar apenas latas de 18 litros: (apenas latas inteiras)

In [10]:
if litros_tinta % 18 > 0:
    latas = litros_tinta // 18 + 1
else:
    latas = litros_tinta // 18
preco = latas * 80
print(f'Serão necessárias {latas} latas, custando R$ {preco} no total')

Serão necessárias 1.0 latas, custando R$ 80.0 no total


##### 2. Comprar apenas galões de 3,6 litros: (apenas galoes inteiros)

In [11]:
if litros_tinta % 3.6 > 0:
    galoes = litros_tinta // 3.6 + 1
else:
    galoes = litros_tinta // 3.6
preco = galoes * 25
print(f'Serão necessários {galoes} galões, custando R$ {preco} no total')

Serão necessários 3.0 galões, custando R$ 75.0 no total


##### 3. Misturar latas e galões, de forma que o desperdício de tinta seja menor. Acrescente 10% de folga e sempre arredonde os valores para cima, isto é, considere latas cheias.

O custo da lata é 80/18 = 4,44 R\\$/L

O custo do galão é 25/3,6 = 6,94 R\\$/L

A lata é mais econômica, então todas as latas inteiras que pudermos usar devemos comprar em latas. Se ficar faltando alguma coisa para completar devemos avaliar se é melhor comprar latas ou galões. Exemplo:

Se queremos comprar 90 litros. 5 latas dão exatamente 90 litros. Então devemos comprar tudo em latas.

Se queremos comprar 95 litros. 5 latas dão exatamente 90 litros. Então devemos comprar pelo menos 5 latas e avaliar o que falta, se estes últimos 5 litros valem mais apenas em latas ou galões.

Para os 5 litros faltantes precisamos de 2 galões que custam 50 reais no total. Ou de uma lata que custa 80 reais no total. Portanto, neste caso vale mais a pena usar 2 galões.

Se queremos comprar 107 litros. 5 latas dão exatamente 90 litros. Então devemos comprar pelo menos 5 latas e avaliar o que falta, se estes últimos 5 litros valem mais apenas em latas ou galões.

Para os 17 litros faltantes precisamos de 5 galões que custam 125 reais no total. Ou de uma lata que custa 80 reais no total. Portanto, neste caso vale mais a pena usar uma lata.

3 galões custam 75 reais, 4 galões custam 100 reais. Então, se for possível completar com até 3 galões escolhe-se galões. Qualquer quantidade maior que 3 galões, usa-se latas.

Podemos ir ao exercício:

In [ ]:
# acrescentamos a folga de 10%
litros_folga = litros_tinta * 1.1
galoes = 0

# só latas inteiras
latas = litros_folga // 18


# aqui vemos a tinta que faltou que não cabe em uma lata completa
tinta_faltante = litros_folga - 18 * latas

# tinta faltante em galões
if tinta_faltante > 0:
    if tinta_faltante % 3.6 > 0:
        faltante_galoes = tinta_faltante // 3.6 + 1
    else:
        faltante_galoes = tinta_faltante // 3.6
    if faltante_galoes <= 3:
        galoes = faltante_galoes
    else:
        galoes = 0
        latas = latas + 1

preco = latas * 80 + galoes * 25
print(f'São necessários {latas} latas e {galoes} galões. Custando R$ {preco} no total')